core/src/autogluon/core/trainer/abstract_trainer.py

```diff
@@ -542,7 +544,16 @@ def stack_new_level(
             aux_models = self.stack_new_level_aux(
                 X=X, y=y, base_model_names=core_models, level=level + 1, infer_limit=infer_limit, infer_limit_batch_size=infer_limit_batch_size, **aux_kwargs
             )
+            # Compute Bagged Weighted Ensemble needed for info leak protection
+            if (core_kwargs.get("ag_args_fit", None) is not None) and core_kwargs["ag_args_fit"].get("stack_info_leak_protection", False) \
+                    and (not last_level):
+                aux_models += self.stack_new_level_aux(
+                    X=X, y=y, base_model_names=core_models, level=level + 1, infer_limit=infer_limit,
+                    infer_limit_batch_size=infer_limit_batch_size, fit_one_model=False, **aux_kwargs
+                )
+                core_kwargs["ag_args_fit"]["reasonable_ensemble"] = self.load_model(aux_models[-1])
```

Comment:
It might make more sense to put `"reasonable_ensemble"` and/or `"stack_info_leak_protection"` as an arg in `"ag_args_ensemble"` rather than `"ag_args_fit"`, that way it is a hyperparameter passed directly to the ensemble itself (such as the BaggedEnsembleModel), rather than to the fold model.